File: mpp-proiect-repository-saboalina-Lab3Java/src/main/java/domain/validators/ZborValidator.java
package domain.validators;

import domain.Zbor;

public class ZborValidator implements  Validator<Zbor> {
    @Override
    public void validate(Zbor entity) throws ValidationException {
        StringBuilder err = new StringBuilder();
        if (entity.getId() <= 0)
            err.append("Id zbor invalid\n");
        if(entity.getDestinatie().length()==0)
            err.append("destinatie zbor invalida\n");
        if(entity.getAeroport().length()==0)
            err.append("aeroport zbor invalid\n");
        if (entity.getNrLocuriDisponibile() <= 0)
            err.append("nr locuri zbor invalid\n");
        if(err.length() > 0)
            throw new ValidationException(err.toString());

    }
}

File: C#/Laborator3/domain/validators/ZborValidator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Laborator3.domain.validators
{
    class ZborValidator : IValidator<Zbor>
    {
        public void Validate(Zbor e)
        {
            StringBuilder err = new StringBuilder();
            if (e.id <= 0)
                err.Append("id zbor invalid\n");
            if (e.destinatie.Length == 0)
                err.Append("destinatie zbor invalida\n");
            if (e.aeroport.Length == 0)
                err.Append("aeroport zbor invalid\n");
            if (e.nrLocuriDisponibile <=0)
                err.Append("nr locuri zbor invalid\n");
            if (err.Length > 0)
                throw new ValidationException(err.ToString());
        }
    }
}
